tests/test_core.py

```diff
@@ -627,10 +627,54 @@ def test_interpolate(test_pd_df):
     assert all([True if isstr(i) else ~np.isnan(i) for i in df.data.foo])
 
 
+def test_interpolate_time_exists(test_pd_df):
+    _df = test_pd_df.copy()
+    df = IamDataFrame(_df)
+    df.interpolate(2005)
+    obs = df.filter(year=2005).data['value'].reset_index(drop=True)
+    exp = pd.Series([1.0, 0.5, 2.0], name='value')
+    pd.testing.assert_series_equal(obs, exp)
+
+
+def test_interpolate_multiple_times(test_pd_df):
+    _df = test_pd_df.copy()
+    df = IamDataFrame(_df)
```

Comment:
As above...

Also suggest to rename the test for clarity
```suggestion
def test_interpolate_with_list(test_df_year):
```